File: Random/twoSum.go
package main

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for i, x := range nums {
		comp := target - x

		if value, flag := m[comp]; flag {
			return []int{value, i}
			// value is actually an index value
		}
		m[x] = i
	}
	return []int{}
}

func alternate(nums []int, target int) []int {
	l := len(nums)
	for i := 0; i < l; i++ {
		for j := i; j < l; j++ {
			if nums[i]+nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}

File: Random/twoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        l = len(nums)
        data = {}
        for i in range(l):
            xa = nums[i]
            if xa in data:
                return [data[xa], i]
            xb = target - xa
            data[xb] = i
        return []
    
    def alternate(self, nums: list[int], target: int) -> list[int]:
        l = len(nums)
        for i in range(l - 1):
            for j in range(i + 1, l):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []
